src/test/moves/roost.test.ts

```diff
@@ -27,33 +27,205 @@ describe("Moves - Roost", () => {
   beforeEach(() => {
     game = new GameManager(phaserGame);
     game.override.battleType("single");
-    game.override.enemySpecies(Species.STARAPTOR);
-    game.override.enemyAbility(Abilities.INSOMNIA);
+    game.override.enemySpecies(Species.RELICANTH);
     game.override.startingLevel(100);
-    game.override.enemyLevel(100);
-    game.override.moveset([Moves.STOMPING_TANTRUM]);
-    game.override.enemyMoveset([Moves.ROOST, Moves.ROOST, Moves.ROOST, Moves.ROOST]);
+    game.override.enemyLevel(60);
+    game.override.enemyMoveset([Moves.EARTHQUAKE, Moves.EARTHQUAKE, Moves.EARTHQUAKE, Moves.EARTHQUAKE]);
+    game.override.moveset([Moves.ROOST, Moves.BURN_UP]);
+    game.override.starterForms({ [Species.ROTOM]: 4 });
   });
 
+  /**
+   * Roost's behavior should be defined as:
+   * The pokemon loses its flying type for a turn. If the pokemon was ungroundd solely due to being a flying type, it will be grounded until end of turn.
+   *
+   * 1. Pure Flying type pokemon -> become normal type until end of turn
+   * 2. Dual Flying/X type pokemon -> become type X until end of turn
+   * 3. Pokemon that use burn up into roost (ex. Moltres) -> become flying due to burn up, then typeless until end of turn after using roost
+   * 4. If a pokemon is afflicted with Forest's Curse or Trick or treat, dual type pokemon will become 3 type pokemon after the flying type is regained
+   *    Pure flying types become (Grass or Ghost) and then back to flying/ (Grass or Ghost),
+   *    and pokemon post Burn up become ()
+   * 5. If a pokemon is also ungrounded due to other reasons (such as levitate), it will stay ungrounded post roost, despite not being flying type.
+   * 6. Non flying types using roost (such as dunsparce) are already grounded, so this move will only heal and have no other effects.
+   *
+   * Also making the user pokemon go first so we can check if the pokemon took damage post roost
+   */
+
+  test(
+    "Non flying type uses roost -> no type change, took damage",
+    async () => {
+      await game.classicMode.startBattle([Species.DUNSPARCE]);
+      game.move.select(Moves.ROOST);
+      const playerPokemon = game.scene.getPlayerPokemon()!;
+      const playerPokemonStartingHP = playerPokemon.hp;
+      let playerPokemonTypes = playerPokemon.getTypes();
```

Comment:
```suggestion
      const playerPokemon = game.scene.getPlayerPokemon()!;
      const playerPokemonStartingHP = playerPokemon.hp;
      let playerPokemonTypes = playerPokemon.getTypes();
      
      game.move.select(Moves.ROOST);
```
This avoids potential desync issues. You're currently collecting the player's types and starting HP as the turn is progressing (this also applies to the other tests)

Reply:
Thanks for all the help!